Adding info about new http client (#4989)
segmentio/segment-docs: src/connections/sources/catalog/libraries/server/node/index.md

`flushInterval` _number_ | The number of milliseconds to wait before flushing the queue automatically. The default is: `10000`
`httpRequestTimeout` | The maximum number of milliseconds to wait for an http request. The default is: `10000`
`disable` | Disable the analytics library for testing. The default is: `false`

> `httpClient` | Optional custom AnalyticsHTTPClient implementation to support alternate libraries or proxies.  Defaults to global fetch or node-fetch for older versions of node.

Reviewer: ```suggestion
`httpClient` *Optional* | A custom AnalyticsHTTPClient implementation to support alternate libraries or proxies.  Defaults to global fetch or node-fetch for older versions of node.
```

const marketingAnalytics = new Analytics({ writeKey: 'MARKETING_WRITE_KEY' });
const appAnalytics = new Analytics({ writeKey: 'APP_WRITE_KEY' });
```
## AnalyticsHTTPClient

> We attempt to use the global `fetch` implementation if available in order to support several diverse environments.  Some special cases (e.g. http proxy) may require a different implementation for http communication.  A customized wrapper can be provided in the Analytics configuration to support this.  Here are a few approaches:

Reviewer: ```suggestion
Segment attempts to use the global `fetch` implementation if available in order to support several diverse environments.  Some special cases (for example, http proxy) may require a different implementation for http communication.  You can provide a customized wrapper in the Analytics configuration to support this.  Here are a few approaches:
```

Use a custom fetch-like implementation with proxy (simple, recommended)

> ```

Reviewer 2: This should be a ```javascript

import { HTTPFetchFn } from '../lib/http-client'

> 

Reviewer 2: Maybe also include `import axios from 'axios'`

const httpClient: HTTPFetchFn = async (url, options) => {
  return axios({
    url,
    proxy: {
        protocol: 'http',
        host: 'proxy.example.com',
        port: 8886,
        auth: {
          username: 'user',
          password: 'pass',
        },
      },
    ...options,
  })
}

> new Analytics({

Reviewer 2: missing "export const analytics = new Analytics..."
